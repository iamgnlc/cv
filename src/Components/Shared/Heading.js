import React from "react"
import PropTypes from "prop-types"

const Heading = React.memo(function Heading(props) {
  return props.label ? (
    <h2
      className={`heading m-0 animated ${
        props.inViewport
          ? global.headingAnimation.in
          : global.headingAnimation.out
      }`}
      dangerouslySetInnerHTML={{ __html: props.label }}
    />
  ) : null
})

Heading.propTypes = {
  label: PropTypes.string.isRequired,
  inViewport: PropTypes.bool.isRequired,
}

export default Heading
