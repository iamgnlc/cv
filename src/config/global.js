global.title = "Curriculum Vitae of Gianluca Agnocchetti"

global.meta = {
  author: "GNLC",
  description: "Curriculum Vitae of Gianluca Agnocchetti",
}

global.brand = "gnlc@localhost ~ $"

global.githubRepo = "https://github.com/iamgnlc/"

global.printHeader =
  '<h1 class="mb-3 text-right">Curriculum Vitae of Gianluca Agnocchetti</h1><h4 class="mb-5 text-right">Senior full stack web developer</h4>'

global.sectionAnimation = {
  in: "fadeIn",
  out: "fadeOut",
}

global.headingAnimation = {
  in: "animation-delay-100 fadeInDown",
  out: null,
}
