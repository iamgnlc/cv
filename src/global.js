global.topOffset = 60

global.title = "Curriculum Vitae of Gianluca Agnocchetti"

global.meta = {
  author: "GNLC",
  description: "Curriculum Vitae of Gianluca Agnocchetti"
}

global.brand = "gnlc@localhost ~Â $"

global.githubRepo = "https://github.com/iamgnlc/cv"
