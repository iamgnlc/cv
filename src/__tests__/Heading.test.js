import React from "react"
import renderer from "react-test-renderer"

import "../config/global.js"

import Heading from "../Components/Shared/Heading"

describe("heading component", () => {
  it("should render without crashing", () => {
    const tree = renderer
      .create(<Heading label="test heading" inViewport={true} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
