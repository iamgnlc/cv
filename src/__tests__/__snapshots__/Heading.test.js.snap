// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`heading component should render without crashing 1`] = `
<h2
  className="heading m-0 animated animation-delay-100 fadeInDown"
  dangerouslySetInnerHTML={
    Object {
      "__html": "test heading",
    }
  }
/>
`;
